Makefile.pre.in

```diff
@@ -927,6 +930,21 @@ SRC_GDB_HOOKS=$(srcdir)/Tools/gdb/libpython.py
 $(BUILDPYTHON)-gdb.py: $(SRC_GDB_HOOKS)
 	$(INSTALL_DATA) $(SRC_GDB_HOOKS) $(BUILDPYTHON)-gdb.py
 
+# Compliance with app stores (such as iOS and macOS) sometimes requires making
+# modifications to the Python standard library. If enabled, apply the patch of
+# known modifications to the source tree before building. The patch will be
+# applied in a dry-run mode (validating, but not applying the patch) on builds
+# that *have* a compliance patch, but where compliance has not been enabled.
+build/app-store-compliant:
+	patch @APP_STORE_COMPLIANCE_PATCH_FLAGS@ --forward --strip=1 --directory="$(srcdir)" --input "$(APP_STORE_COMPLIANCE_PATCH)" ; \
+	if test "@APP_STORE_COMPLIANCE_PATCH_FLAGS@" == ""; then \
+		echo "Source tree has been patched for app store compliance."; \
+		mkdir -p build ; \
+		echo "$(APP_STORE_COMPLIANCE_PATCH)" > build/app-store-compliant ; \
+	else \
+		echo "App store compliance patch has been verified, but not applied."; \
```

Comment:
We often (but not always) silence echoing using `@` in the Makefile. Is it worth it here? (I guess we'd have to silence the whole `if`, since it's technically all one line.)

Reply:
Makes sense - you can work out whether the patch has been applied from the `patch` command in the log, and the existence of the marker file; no need for the extra noise.

Reply:
Oh, I didn't mean removing the `echo` commands, just silencing the _echoing_ of the script by using `@` :) Anyway, I'm fine with 87a8e72!